examples/pybadger_tilegame.py

```diff
@@ -0,0 +1,527 @@
+            if x_offset != 0 or y_offset != 0:
+                # variable to store if player is allowed to move
+                can_move = False
+
+                # coordinates the player is moving to
+                moving_to_coords = (GAME_STATE['PLAYER_LOC'][0] + x_offset, GAME_STATE['PLAYER_LOC'][1] + y_offset)
+
+                # tile name of the spot player is moving to
+                moving_to_tile_name = GAME_STATE['CURRENT_MAP'][moving_to_coords[0], moving_to_coords[1]]
+
+                # if there are entity(s) at spot the player is moving to
+                if moving_to_coords in GAME_STATE['ENTITY_SPRITES_DICT']:
+                    print("found entity(s) where we are moving to")
+
+                    # loop over all entities at the location player is moving to
+                    for entity_obj in GAME_STATE['ENTITY_SPRITES_DICT'][moving_to_coords]:
+                        print("checking entity %s" % entity_obj["map_tile_name"])
+                        # if the entity has a before_move behavior function
+                        if "before_move" in TILES[entity_obj["map_tile_name"]].keys():
+                            print("calling before_move %s, %s, %s" % (
+                                moving_to_coords, GAME_STATE['PLAYER_LOC'], entity_obj))
+                            # call the before_move behavior function act upon it's result
+                            if TILES[entity_obj["map_tile_name"]]['before_move'](moving_to_coords,
+                                                                                 GAME_STATE['PLAYER_LOC'], entity_obj,
+                                                                                 GAME_STATE):
+                                # all the movement if it returned true
+                                can_move = True
+                            else:
+                                # break and don't allow movement if it returned false
+                                break
+                        else:  # entity does not have a before_move function
+                            # allow movement
+                            can_move = True
+                    if can_move:
+                        # set the player loc variable to the new coords
+                        GAME_STATE['PLAYER_LOC'] = moving_to_coords
+
+                else:  # no entities at the location player is moving to
+                    # set player loc variable to new coords
+                    GAME_STATE['PLAYER_LOC'] = moving_to_coords
+
+            # reset movement offset variables
+            y_offset = 0
+            x_offset = 0
+
+            # set previos button values for next iteration
+            prev_up = cur_up
+            prev_down = cur_down
+            prev_right = cur_right
+            prev_left = cur_left
+
+            # current time
+            now = time.monotonic()
+
+            # if it has been long enough based on FPS delay
```

Comment:
Have you tried using displayio's manual refresh mode by turning auto_refresh off? It's `refresh` takes in a target frame rate which will slow down if the code is too fast and skip the frame if the code was too slow.

Reply:
Oh, neat. I did not know about that. Thank you.